import-scripts/import-genie-data-transfer-deployment-color.sh

```diff
@@ -0,0 +1,209 @@
+        else
+            echo "Error : invalid argument '$DEPLOYMENT_COLOR' passed to all_replicas_ready()" >&2
+            exit 1
+        fi
+    fi
+    headers_read=0
+    while IFS= read -r line; do
+        if [ $headers_read -eq 0 ] ; then
+            headers_read=1
+        else
+            parse_line_regex='^([[:graph:]]*)[[:space:]]*([[:digit:]]*)/([[:digit:]])*.*'
+            if ! [[ $line =~ $parse_line_regex ]] ; then
+                echo "Error : failure to parse format of get deployment output : $line" >&2
+                exit 1
+            fi
+            deployment_name=${BASH_REMATCH[1]}
+            ready_pods=${BASH_REMATCH[2]}
+            total_pods=${BASH_REMATCH[3]}
+            if [ $ready_pods -ne $total_pods ] ; then
+                rm -f $DEPLOYMENT_CHECK_OUTPUT_FILEPATH
+                return 1 # not all ready yet
+            fi
+        fi
+    done < $DEPLOYMENT_CHECK_OUTPUT_FILEPATH
+    rm -f $DEPLOYMENT_CHECK_OUTPUT_FILEPATH
+    return 0 # all ready
+}
+
+function scale_deployment_to_N_replicas() {
+    DEPLOYMENT_COLOR=$1
+    NUM_REPLICAS=$2
+    full_minus_1=$(($FULL_REPLICA_COUNT-1))
+    if [ $DEPLOYMENT_COLOR == 'blue' ] ; then
+        kubectl --kubeconfig $PUBLIC_CLUSTER_KUBECONFIG scale deployment --replicas $NUM_REPLICAS ${GENIE_BLUE_DEPLOYMENT_LIST[@]}
+    else
+        if [ $DEPLOYMENT_COLOR == 'green' ] ; then
+            kubectl --kubeconfig $PUBLIC_CLUSTER_KUBECONFIG scale deployment --replicas $NUM_REPLICAS ${GENIE_GREEN_DEPLOYMENT_LIST[@]}
+        else
+            echo "Error : invalid argument '$DEPLOYMENT_COLOR' passed to scale_deployment_to_full_minus_1_replicas()" >&2
+            exit 1
+        fi
+    fi
+    # now wait for replicas to become ready
+    check_count=0
+    while [ $check_count -lt $REPLICA_READY_CHECK_MAX_CHECKCOUNT ] ; do
+        sleep $REPLICA_READY_CHECK_PAUSE_SECONDS
+        if all_replicas_ready $DEPLOYMENT_COLOR ; then
+            return 0
+        fi
+        check_count=$(($check_count+1))
+    done
+    echo "Error : scaling for '$DEPLOYMENT_COLOR' deployments failed to reach the ready state after $REPLICA_READY_CHECK_MAX_CHECKCOUNT iterations with a period of $REPLICA_READY_CHECK_PAUSE_SECONDS seconds. Exiting"
+    exit 1
+}
+
+function switchover_ingress_rules_to_updated_database_deployment() {
+    # recently we have checked that the k8s config repo is up to date and matches the production specs
+    # TODO switch over ingress rules to the new deployment
+    # - alter selected services in ingress rules for genie.cbioportal.org and genie-private.cbioportal.org in cbio-ingress.yaml
+    echo "switching traffic over to the updated database deployment"
+}
+
+function check_in_changes_to_kubernetes_into_github() {
+    #TODO: implement commit and push .. report problems as error
+    echo "checking in configuration changes to github"
+}
+
+function main() {
```

Comment:
the main function contains a lot of different steps - I think readability could be improved by separating the different steps with comments (perhaps using the comments from the file header) / adding some whitespace in between